index.d.ts

```diff
@@ -0,0 +1,208 @@
+/// <reference types="node" />
+/// <reference types="aws-lambda" />
+/// <reference types="connect" />
+
+import { IncomingMessage, ServerResponse } from 'http';
+import { Handler } from 'aws-lambda';
+import { ErrorHandleFunction } from 'connect';
+
+export default agent;
```

Comment:
This would mean that users would need to make typing work in this way, 

<img width="619" alt="Screenshot 2019-03-20 at 11 41 00" src="https://user-images.githubusercontent.com/3902525/54678289-0efea580-4b05-11e9-89df-0efd745dfd4e.png">

I think we should do this instead 
```ts
export = Agent
```
<img width="639" alt="Screenshot 2019-03-20 at 11 42 01" src="https://user-images.githubusercontent.com/3902525/54678332-32295500-4b05-11e9-857f-12c130998d95.png">

Reply:
If we want to make other exports work, everything should be namespaced inside Agent and we can do 
```ts
export = Agent
export as namespace Agent

declare namespace Agent {

}
```

Reply:
I think you got the API of the module wrong. We're not exporting the `Agent` object, but an instance of that object

Reply:
🤔 though you are correct about something not working regarding the export... not really sure what's wrong though.

I thought `export default foo` was the same as `module.exports = foo`. But apparently not 100%

Reply:
How would you write the CommonJS style `module.exports = foo` as an ES Modules export?

Reply:
> I thought export default foo was the same as module.exports = foo. But apparently not 100%

I am not sure though, its node's internal resolution. I guess this is the equivalent of `export default foo`
```js
module.exports = {
  __esModule: true,
  default: foo
}
```

Reply:
The reason because import works and require doesn't is because in ES6, the default export is also a named export and `default` is a reserved keyword. Since we have typescript definition in ES6 export and my example was importing it as a common js, the default export has to be explicitly used. But if we used `import agent from 'elastic-apm-node'` it would work simply since the default export would be resolved properly.

Reply:
As suggested before `export = Agent` and using namespaces we can guarantee it works both on commonjs and es6 modules. More details here http://www.typescriptlang.org/docs/handbook/declaration-files/deep-dive.html

Reply:
I read the entire thing but only got more confused 😅

What did you say was the way to express `module.exports = new Agent()` in a `.d.ts` file?

Reply:
What you're already doing is the correct way to export an instance of a thing.

```ts
declare const agent: Agent;
export default agent;
```

Reply:
I thought so as well. The problem is just that it only works if the user is using ES Modules. If the user is using CommonJS, Intellisense wrongly thinks that they need to import the module like so for it to be valid:

```js
const agnet = require('elastic-apm-node').default
```

Reply:
And while we could make a `default` property available on the agent object that pointed to the agent object it self, that would just be a hack to make the typings happy, which I think is kind of silly

Reply:
Oh, actually `export default agent` maps to the ESM `default` value, while `export = agent` would map directly to a commonjs module export.

Reply:
Ha, @sqren just told me the same thing. @vigneshshanmugam actually also wrote it above, but I got confused about the namespace stuff in the same comment so I never got it to work. But now it works locally. The only downside is that you can't export other things if you use the `export = agent` syntax. Which means that we can't export our types. But I can see from a technical perspective why that is and I guess there's no way around that. So I'll update it and remove all our other exports

Reply:
Fixed